task-sdk/src/airflow/sdk/execution_time/supervisor.py

```diff
@@ -523,24 +532,26 @@ def _register_pipe_readers(self, stdout: socket, stderr: socket, requests: socke
         # alternatives are used automatically) -- this is a way of having "event-based" code, but without
         # needing full async, to read and process output from each socket as it is received.
 
-        # Track socket types for debugging
-        self._fd_to_socket_type = {
-            stdout.fileno(): "stdout",
-            stderr.fileno(): "stderr",
-            requests.fileno(): "requests",
-            logs.fileno(): "logs",
-        }
+        # Track the open sockets, and for debugging what type each one is
+        self._open_sockets.update(
```

Comment:
`update` because this is a WeakKeyDict -- meaning the dict won't keep things open/alive if the socket object gets GCd.